CHANGELOG.md

```diff
@@ -2,160 +2,178 @@
 
 ## Changes Since Last Release
 
+### OCI-mode
+
+Singularity 4 introduces OCI-mode as a fully supported feature. It is enabled by
+using the `--oci` flag with the `run / shell / exec / pull` commands, or by
+setting `oci mode = yes` in `singularity.conf`.
+
+In OCI-mode:
+
+- Container images from OCI sources will be `pull`-ed to an OCI-SIF file. An
+  OCI-SIF file encapsulates the OCI image configuration and squashed filesystem
+  using an OCI, rather than Singularity specific, structure.
+- The `run / shell / exec` commands use a low-level OCI runtime (crun/runc) for container
+  execution.
+- Default operation is compatible with other OCI tools, similar to using
+  `--compat` in Singularity's non-OCI native mode.
+- OCI-modes support running existing Singularity non-OCI-SIF images, and can be
+  made to imitate native mode default behavior by using the `--no-compat` flag.
+
+OCI-mode changes from 3.11 to 4.0 include:
+
+- `run / shell / exec` in OCI-mode now includes support for the following
+  existing CLI flags:
+  - `--add-caps`
+  - `--drop-caps`
+  - `--keep-privs`
+  - `--no-privs`
+  - `--overlay` from directories, bare squashfs and extfs images.
+  - `--workdir`
+  - `--scratch`
+  - `--no-home`
+  - `--no-mount` (dev cannot be disabled in OCI mode
+  - `--no-umask` (with `--no-compat`)
+  - `--writable-tmpfs` (with `--no-compat`)
+- Added `--device` flag to "action" commands (`run`/`exec`/`shell`) when run in
+  OCI mode (`--oci`). Currently supports passing one or more (comma-separated)
+  fully-qualified CDI device names, and those devices will then be made
+  available inside the container.
+- Added `--cdi-dirs` flag to override the default search locations for CDI
+  json files, allowing, for example, users who don't have root access on their
+  host machine to nevertheless create CDI mappings (into containers run with
+  `--fakeroot`, for example).
+- A container run as root, or with `--fakeroot`, has OCI default
+  effective/permitted capabilities.
+- An `--env-file` is evaluated with respect to the host environment, to match
+  native mode behaviour.
+- If the kernel does not support unprivileged overlays, OCI-mode will attempt to
+  use `fuse-overlayfs` and `fusermount` for overlay mounting and unmounting.
+- Support for thee `SINGULARITY_CONTAINLIBS` env var, to specify libraries to
+  bind into `/.singularity.d/libs/` in the container.
+- Support for running OCI-SIF images directly from `docker://`, `http://`,
+  `https://` and `oras://`. URIs.
+- A new `--no-compat` flag can be used with OCI-mode to reduce OCI runtime
+  compatibility and emulate singularity's historic native mode behaviour:
```

Comment:
Don't love the phrase "to reduce OCI runtime compatibility" without specifying compatibility _with what_. (I presume what is meant was "... with other OCI runtimes"; but who knows what a reader may insert when reading this.) What about: "A new `--no-compat` flag can be used with OCI-mode to mirror singularity's historic native mode behavior on a variety of settings, instead of setting them the way other OCI runtimes typically do:"